### PR TITLE
#33 creating comments

### LINKED ISSUE #47: rubocop user{ nil }
DEPRECATION WARNING: Static attributes will be removed in FactoryBot 5.0. Please use dynamic

attributes instead by wrapping the attribute value in a block:

user { nil }

To automatically update from static attributes to dynamic ones,
install rubocop-rspec and run:

rubocop \
  --require rubocop-rspec \
  --only FactoryBot/AttributeDefinedStatically \
  --auto-correct

### LINKED ISSUE #48: rubocop message { "Test post" }
message { "Test post" }

To automatically update from static attributes to dynamic ones,
install rubocop-rspec and run:

rubocop \
  --require rubocop-rspec \
  --only FactoryBot/AttributeDefinedStatically \
  --auto-correct

 (called from block (2 levels) in <top (required)> at /Users/toddlangford-archer/Documents/makers/week-8+9/acebook-stars/spec/factories/posts.rb:4)

### LINKED ISSUE #50: user id must match current user id when editing comments/posts

### PR DESCRIPTION
**Pull request by toddpla**
fixes #48 
fixes #47 
fixes #50

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -221,6 +221,7 @@ GEM
     thor (0.20.0)
     thread_safe (0.3.6)
     tilt (2.0.8)
+    timecop (0.9.1)
```
Comment (runticle) [inline Gemfile.lock]: watch out bernard

File: spec/controllers/comments_controller_spec.rb [hunk 2]
```diff
@@ -1,5 +1,48 @@
 require 'rails_helper'
-
+require 'pry'
```
Comment (imogenmisso) [inline spec/controllers/comments_controller_spec.rb]: We can delete this

Comment (rsmith88) [inline spec/controllers/comments_controller_spec.rb]: Should 'require pry' before merging with master?

Comment (rsmith88) [reply spec/controllers/comments_controller_spec.rb]: Sorry that should say 'be removed before merging with master'

File: spec/controllers/comments_controller_spec.rb [hunk 3]
```diff
@@ -1,5 +1,48 @@
...
+  end
+
+  describe 'POST /' do
+    it 'responds with 200' do
+      post :create, params: {comment: {comment: 'Nice comment' }, post_id: @post.id}
+      expect(response).to redirect_to(posts_url)
+    end
+
+    it 'creates a comment' do
+      post :create, params: {comment: {comment: 'Newly created comment' }, post_id: @post.id}
```
Comment (runticle) [inline spec/controllers/comments_controller_spec.rb]: whats this syntax?

File: spec/controllers/comments_controller_spec.rb (hunk 2 above)
Comment (runticle) [reply spec/controllers/comments_controller_spec.rb]: Delete.

File: spec/controllers/comments_controller_spec.rb (hunk 3 above)
Comment (imogenmisso) [reply spec/controllers/comments_controller_spec.rb]: Not sure

**Pull request merged**